x-pack/plugins/security/server/lib/get_user.ts

```diff
@@ -4,12 +4,26 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { Request } from 'hapi';
+// @ts-ignore
 import { getClient } from '../../../../server/lib/get_client_shield';
 
-export function getUserProvider(server) {
+interface UserRealm {
+  name: string;
+  type: string;
+}
+
+export interface User {
```

Comment:
__note:__ just defined a couple of props we need or may need soon.

Reply:
**note** I'm also introducing a User interface as part of #30977 (https://github.com/elastic/kibana/pull/30977/files#diff-31444acbfefac2d86a416260887ab8cbR7)

Whoever merges last should try to consolidate the interfaces

Reply:
Ha, good to know! Any reason why `authentication_realm` and `lookup_realm` are optional?

Reply:
I marked them optional because we don't always have that information in all UI contexts. We only get `authentication_realm` and `lookup_realm` for the currently logged in user, but pages like the user management screen allow you to view/create/update arbitrary users, and that information isn't available nor necessary then.

If it's easier/better for your use case to mark them required, then maybe we end up having a couple of interfaces, where one extends the other? Maybe something like `User` and `AuthenticatedUser`?

Reply:
Yeah, I think these two are different interfaces even though they look similar. The one I use is more like `AuthenticatedUser\Principal\Identity`.